rust/src/types/execution_context.rs

```diff
@@ -0,0 +1,121 @@
+use evmc_vm::{
+    AccessStatus, Address, ExecutionContext, ExecutionMessage, ExecutionResult, ExecutionTxContext,
+    StorageStatus, Uint256,
+};
+
+#[cfg_attr(test, mockall::automock)]
```

Comment:
Does this macro define `MockExecutionContextTrait`? 

does it define it for `test` configuration or for all of them ?

Reply:
yet it does but only if the mock feature is enabled (which is the case for tests)